tests/e2e/bq-batch-retrieval.py

```diff
@@ -509,17 +511,14 @@ def test_update_featureset_update_featureset_and_ingest_second_subset(
     )
     update_fs.infer_fields_from_df(subset_df)
     client.apply(update_fs)
+    client.ingest(feature_set=update_fs, source=subset_df)
+
     while True:
-        fs = client.get_feature_set(name=update_fs.name, project=update_fs.project)
-        if fs.status == FeatureSetStatus.STATUS_READY:
-            # Give BQ table schema some time to update
-            time.sleep(120)
+        buf_size = get_streaming_buffer_size(client, update_fs)
+        if buf_size == 10:
             break
-        time.sleep(5)
+        time.sleep(15)
```

Comment:
This is very nit picky, but I am thinking of removing these sleeps over time and adding timeouts/retries. Inevitably the sleeps just keep getting bumped up and up over time.

Should we maybe start incorporating https://github.com/rholder/retrying?

Reply:
I guess you dont have to be the one to do it, but if you end up touching the sleeps again then I think it would be nice to consider it.

Reply:
This is a proper retry with a 15s backoff though. I havent touched the other sleeps.